runtime/onert/core/src/ir/OpSequences.cc

```diff
@@ -30,12 +30,46 @@ OpSequenceIndex OpSequences::emplace(const OperationIndex &index, Layout layout)
 {
   std::unique_ptr<OpSequence> op_seq = std::make_unique<OpSequence>(layout);
   op_seq->appendOperation(index);
-  return push(std::move(op_seq));
+  const OpSequenceIndex &seq_index = push(std::move(op_seq));
+  cacheSequenceIndex(seq_index, index);
+  return seq_index;
 }
 
 OpSequenceIndex OpSequences::emplace(std::unique_ptr<OpSequence> &&op_seq)
 {
-  return push(std::move(op_seq));
+  auto &operations = op_seq->operations();
+  const OpSequenceIndex &seq_index = push(std::move(op_seq));
+  for (const auto &op_idx : operations)
+  {
+    cacheSequenceIndex(seq_index, op_idx);
+  }
+  return seq_index;
+}
+
+void OpSequences::cacheSequenceIndex(const OpSequenceIndex &seq_index,
+                                     const OperationIndex &op_index) const
+{
+  _seq_indexes.insert(
+      std::pair<const OperationIndex &, const OpSequenceIndex &>(op_index, seq_index));
+}
+
+OpSequenceIndex *OpSequences::findSequenceIndex(const OperationIndex &operation_index) const
+{
+  // If opration_index is cached, return sequence_index from cache
+  if (_seq_indexes.count(operation_index))
+  {
+    auto &op_seq_index = _seq_indexes.at(operation_index);
+    if (_objects.count(op_seq_index) && _objects.at(op_seq_index)->exist(operation_index))
+    {
+      return &op_seq_index;
+    }
+    else
+    {
+      _seq_indexes.erase(operation_index);
```

Comment:
@periannath What about making a function for erasing the cache slot like `cacheSequenceIndex`?

Reply:
@periannath As we talked offline, I would like to introduce nested class `Cache` with `add`, `remove`, `find`. Instead of `cacheSequenceIndex`, `_seq.indexes.erase(...)` and `findSequenceIndex`.

Reply:
@glistening I have thought about it, but I can't find a proper name for it.